File: 2375-construct-smallest-number-from-di-string/2375-construct-smallest-number-from-di-string.java
class Solution {
    public String smallestNumber(String pattern) {
        StringBuilder sb1 = new StringBuilder();
        StringBuilder sb2 = new StringBuilder();
        char numChar = '1';
        char[] chars = pattern.toCharArray();
        for (char c : chars) {
            sb2.append(numChar++);
            if (c == 'I') {
                sb1.append(sb2.reverse());
                sb2.setLength(0);
            }
        }
        sb2.append(numChar++);
        sb1.append(sb2.reverse());
        return sb1.toString();
    }
}

File: 2375-construct-smallest-number-from-di-string/2375-construct-smallest-number-from-di-string.kt
class Solution {
    fun smallestNumber(pattern: String): String {
        val res = StringBuilder()
        var temp = StringBuilder()
        var ch = '1'
        for (i in 0 until pattern.length) {
            val cur = pattern[i]
            temp.append(ch++)
            if (cur == 'I') {
                res.append(temp.reverse())
                temp = StringBuilder()
            }
        }
        temp.append(ch++)
        res.append(temp.reverse())
        return res.toString()
    }
}